### PR TITLE
APM-5185_Proxygen_cli_Ne_Changes

### PR DESCRIPTION
**Pull request by NautiyalVikas1**

### CODE REVIEW

**Review by acarriedev — CHANGES_REQUESTED** [review]
I've left some comments to address. 

I appreciate we need to merge it to test that the env vars coming in from the actions will work for the packages but I think the functionality won't work how we expect it i.e. being able to overwrite the changes.

You haven't currently added any tests so I think adding tests to prove out the functionality from the acceptance criteria should help.

File: README.md [hunk 1]
```diff
@@ -15,18 +15,23 @@ After installation, the `proxygen` executable is available. Typing `proxygen` di
 
 The CLI requires client credentials, which can be obtained from the `platforms-api-producer-support` Slack channel.
 
-All users should also have individual credentials. `proxygen-cli` needs to know about them.
-You can setup the credentials in `proxygen-cli` using:
+The Proxygen CLI client credentials are setup now as part of proxygen cli package.
```
Comment (acarriedev) [inline README.md:18]: I think would be better to just explain it like it is currently. Rarther than saying 'Now its like this' could we reword it to be more 'It is like this'

File: README.md [hunk 2]
```diff
@@ -15,18 +15,23 @@ After installation, the `proxygen` executable is available. Typing `proxygen` di
...
+
+if in case client credentials are not setup as part of package installation or you want to update  the credentials in the future, use: 
+
 proxygen credentials set
 ```
-This command prompts you to enter your `client_id`, `client_secret`, `username`, and `password`. 
-To update credentials in the future, use:
+This command prompts you to enter your `client_id`, `client_secret`, `username`, and `password`.
+and also this will help to update the default client credentials (`client_id`, `client_secret`) in the future, use:
 ```
```
Comment (acarriedev) [inline README.md:31]: This is good for your current ticket but it seems the README isn't clear on the difference between user access and client access. Maybe the package isn't quite configured right for the 2 as well. Maybe we can have a think about this and create a follow on ticket if we're not happy as it would be out of scope

File: proxygen_cli/cli/command_credentials.py [hunk 3]
```diff
@@ -58,20 +60,27 @@ def set(custom_pairs, force):
...
-        current_credentials["client_secret"] = client_secret
         current_credentials["username"] = username
         current_credentials["password"] = password
 
+    # Check if client credentials are set
+    client_credentials_set = all(
+        current_credentials.get(field) is not None
+        for field in ["client_id", "client_secret"]
+    )
+    if not client_credentials_set or force:
```
Comment (acarriedev) [inline proxygen_cli/cli/command_credentials.py:78]: Does this still mean force will always ask for client id and secret?

File: proxygen_cli/test/command_credentials_test.py [hunk 4]
```diff
@@ -26,6 +28,19 @@ def get_test_credentials(**kwargs):
 
     return "\n".join(text_format_credentials)
 
+def get_test_client_credentials(**kwargs):
```
Comment (acarriedev) [inline proxygen_cli/test/command_credentials_test.py:31]: Ideally we need to cover more scenarios. Should be multiple tests around setting credentials - 
- are the defaults set but we still need to provide username/password
- If we force overwrite username/password can we do that without knowing client id secret
- can we overwrite client id secret if we want to

**Review by acarriedev — CHANGES_REQUESTED** [review]

File: proxygen_cli/cli/command_credentials.py (hunk 3 above)
Comment (NautiyalVikas1) [reply proxygen_cli/cli/command_credentials.py:78]: Yes Alex We need to remove force as we want to this happen only when Client Credenntials (Client_Id and Client_secret) are not set as part of package or need update.

File: README.md (hunk 2 above)
Comment (NautiyalVikas1) [reply README.md:31]: Sure Alex Will go through it and I think we need Follow up ticket on this.

File: proxygen_cli/test/command_credentials_test.py (hunk 4 above)
Comment (NautiyalVikas1) [reply proxygen_cli/test/command_credentials_test.py:31]: Working on Adding more test cases at the moment.

File: README.md (hunk 1 above)
Comment (NautiyalVikas1) [reply README.md:18]: Sure Alex I will make changes and Update the PR

**Review by acarriedev — APPROVED** [review]
